src/index.js

```diff
@@ -30,11 +30,30 @@ class Container extends Component {
   }
 
   prepareInitialStore = (options = {}, prevOptions = {}) => {
-    const { userDetails: { smsNumber } = {} } = options
-    const { userDetails: { smsNumber: prevSmsNumber } = {} } = prevOptions
+    const { userDetails: { smsNumber } = {}, steps } = options
+    const { userDetails: { smsNumber: prevSmsNumber } = {}, steps: prevSteps } = prevOptions
 
-    if (smsNumber && smsNumber !== prevSmsNumber)
+    if (smsNumber && smsNumber !== prevSmsNumber) {
       actions.setMobileNumber(smsNumber)
+    }
+
+    if (steps && steps !== prevSteps) {
+      this.handledPreselectedDocument(steps, actions)
+    }
+  }
+
+  handledPreselectedDocument = (steps, actions) => {
+    const isDocument = (step) => step.type === 'document'
+    const documentStep = Array.find(steps, isDocument)
+    const hasDocTypes = documentStep && documentStep.options && documentStep.options.documentTypes
+    if (hasDocTypes) {
+      const docTypes = documentStep.options.documentTypes
```

Comment:
`hasDocTypes` is the same as `docTypes` inside this `if`. Might as well use it, but if so it should be renamed.